File: 1047.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack=[]
        for i in s:
            if len(stack)==0:
                stack.append(i)
                continue
            top=stack[-1]
            stack.append(i)
            if top==stack[-1]:
                stack.pop()
                stack.pop()
            
        return "".join(stack)

File: 1047.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        vector<char> stack;
        char top;
        string result;
        for(int i=0;i<s.length();i++)
        {
            if(!stack.empty())
            {
                top=stack.back();
                stack.push_back(s[i]);
                if(top==stack.back())
                {
                    stack.pop_back();
                    stack.pop_back();
                }
            }
            else
            {
                stack.push_back(s[i]);
            }
        }
        for(int i=0;i<stack.size();i++)
        {
            result.push_back(stack[i]);
        }
        return result;
    }
};
